app/controllers/posts_controller.rb

```diff
@@ -1,7 +1,7 @@
 class PostsController < ApplicationController
   def index
-    @user = User.find(params[:user_id])
-    @posts = @user.posts
+    @user = User.includes(:posts).find(params[:user_id])
```

Comment:
[OPTIONAL] It would have been better if you had done something like this: 
`@posts = @user.posts.includes(:comments)`. It would be faster.
Your code fetches  users then includes all posts then finds a specific user and its posts